### DISCUSSION TITLE
How to not get repetitive PDB outputs while using RosettaDNA application?

### DISCUSSION THREAD

**Discussion by intomybioverse** [RosettaCommons/rosetta#238] [Rosetta] [answered]
Hello everyone,
While using the RosettaDNA application, I am getting repetitive PDB structures as output. More than 90% of the generated models have the same mutations in the interface residues. The minute differences in the binding energy and total energy are due to the differences in conformational states in the output PDB structures. 

How can I get maximum number of possible mutations for the targeted interface residues that increase the specificity of the protein towards the target DNA molecule?

Here is my resfile.flags:

**resfile.flags:**
-mute all
-unmute protocols.dna
-in:ignore_unrecognized_res
-score:weights dna
-use_input_sc
-run:output_hbond_info
-score:output_residue_energies
-run:min_type dfpmin_armijo
-run:min_tolerance 0.0001
-adducts dna_major_groove_water
-ex1
-ex2
-ex3
-ex4
-ex1aro::level 6
-ex2aro::level 6
-nstruct 1000
-jd2:dd_parser
-overwrite
-out:file:scorefile score.sc

**Comment by smlewis** [comment, marked as answer]
1) If you are using multiple Rosetta processes (via MPI, or manually) - check that your RNG seeds vary.  I think this is NOT your problem (you report some variance, and energy variance even when the sequences match - that suggests it's not RNG matching).  These are reported at the top of log files.

2) Frankly this sounds like a tremendously SUCCESSFUL design run.  Convergence like this is usually a sign that everything is going well and lots of trajectories agree on what a good sequence would be.  This is especially true if your design space is small (few positions can mutate).

The packer is not really designed for giving back wide variance answers: convergence is success, so you are kind of asking it to intentionally fail.  In some cases you can do this by intentionally foreshortening the packing steps or increasing the Monte Carlo at which packing occurs, but I have no idea how well this interacts with the DNA scorefunction.

I think the simplest (not necessarily least-time) route to more variation is backbone variety in the inputs.  Try relaxing it (especially sloppily at a higher temperature) and use that ensemble as inputs to this step to see if you get more variation.  (disclaimer, I have no idea if relaxing DNA, especially mucking with the MC temperature, is a good idea)
